scalyr_agent/configuration.py

```diff
@@ -124,6 +124,10 @@ def parse(self):
                 self.__add_elements_from_array('monitors', content, self.__config)
                 self.__merge_server_attributes(fp, content, self.__config)
 
+            # Check for api_key in the environment variable `scalyr_api_key`. Sometimes, injecting a secret key like
+            # this is preferable via environment variables
+            api_key = os.environ.get('scalyr_api_key') if os.environ.get('scalyr_api_key') else api_key
```

Comment:
Hmm... I'd rather only take the key from the environment if none is specified in the configuration file.  So, I think here you have to check to see if the api_key is none.

Reply:
ok, makes sense. This avoids the case of a hierarchy, and there can only be one place where the key can be configured. Would you like to notify the user if both the config file and the environment variable have it set?

Reply:
Yes, that is a good idea.  If both are set, we use the configuration file (because that's preserves the current behavior), but emit a log warning to the agent.log that the environment key is being ignored.